### DISCUSSION TITLE
I want my users to be able to confirm their emails at a later time, but disabling the "Confirm email" option in settings creates users with their confirmation email data set.

### DISCUSSION THREAD

**Discussion by mypalvikram** [supabase/supabase#8197] [Questions] [answered]
Hey, everyone. I'm developing a mobile application and my ideal UX is that users can sign up via email but confirm the email at a later time. Currently, the way to do that is by toggling the "Confirm email" in email auth settings to off. By doing this, when a user signs up, their object key "email_confirmed_at" is populated with a timestamp. 

```
  const { user, error } = await supabase.auth.signUp({
    email: email,
    password: password,
  });
```

Returns a user object with "email_confirmed_at" populated:

` ..., "email_confirmed_at": "2022-08-10T18:57:41.474904745Z", ...`

**How I imagined toggling email confirmations off would work:**
If the email confirmation is toggled off and a user signs up, this key would stay null. It can then be used to notify the user they need to confirm their email before, for example, sending a message or signing up for a service paid service.

Is there something I'm missing on how to handle email confirmation at a later time? Thanks.

**Comment by GaryAustin1** [comment, marked as answer]
You are probably going to need to provide more information to get some help on this.  It is unclear what your goals are.

If you set confirm email on, your user will get an email with a link to confirm their email.  You can set how long they have to do this.

It seems you need more than that, but it is unclear how you would propose to confirm their email at a later time and what you want them to be able to do until they are confirmed.
